core/vm/instructions.go

```diff
@@ -891,11 +891,22 @@ func opUndefined(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) (
 func opStop(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
 	return nil, errStopToken
 }
-
+func opSetIndestructible(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
+	// map initialization
+	if interpreter.evm.indestructibleContracts == nil {
+		interpreter.evm.indestructibleContracts = make(map[common.Address]struct{})
+	}
+	interpreter.evm.indestructibleContracts[scope.Contract.Address()] = struct{}{}
+	return nil, nil
```

Comment:
Per EIP: 
The intended use case would be for contracts to make **their first byte** of code be the SET_INDESTRUCTIBLE opcode.

The way you could do this is to check `pc` AKA program counter, it should be `0`.